pkg/mattermost/mattermost_v1beta.go

```diff
@@ -142,6 +170,70 @@ func GenerateIngressV1Beta(mattermost *mmv1beta.Mattermost) *networkingv1.Ingres
 	return ingress
 }
 
+func GenerateALBIngressClassV1Beta(mattermost *mmv1beta.Mattermost) *networkingv1.IngressClass {
+	ingressClass := &networkingv1.IngressClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "alb",
+			Namespace:       mattermost.Namespace,
+			Labels:          mattermost.MattermostLabels(mattermost.Name),
+			OwnerReferences: MattermostOwnerReference(mattermost),
+		},
+		Spec: networkingv1.IngressClassSpec{
+			Controller: "ingress.k8s.aws/alb",
+		},
+	}
+
+	return ingressClass
+}
+
+// GenerateIngressALBIngressV1Beta returns the AWS ALB ingress for the Mattermost app.
+func GenerateALBIngressV1Beta(mattermost *mmv1beta.Mattermost) *networkingv1.Ingress {
+	ingressAnnotations := map[string]string{}
+
+	if mattermost.Spec.AWSLoadBalancerController.InternetFacing {
+		ingressAnnotations["alb.ingress.kubernetes.io/scheme"] = "internet-facing"
+	} else {
+		ingressAnnotations["alb.ingress.kubernetes.io/scheme"] = "internal"
+	}
+
+	if mattermost.Spec.AWSLoadBalancerController.CertificateARN != "" {
+		ingressAnnotations["alb.ingress.kubernetes.io/certificate-arn"] = mattermost.Spec.AWSLoadBalancerController.CertificateARN
+		ingressAnnotations["alb.ingress.kubernetes.io/ssl-redirect"] = "443"
+		ingressAnnotations["alb.ingress.kubernetes.io/listen-ports"] = `'[{"HTTP": 80}, {"HTTPS":443}]'`
```

Comment:
This gets converted to 

```
    alb.ingress.kubernetes.io/listen-ports: '''[{"HTTP": 80}, {"HTTPS":443}]'''
```
and then Kubernetes complains about the extra quotes.. Is there a work around for this?

Reply:
fixed this in b479418